envoy/upstream/cluster_manager.h

```diff
@@ -431,6 +431,16 @@ class ClusterManager {
   allocateOdCdsApi(const envoy::config::core::v3::ConfigSource& odcds_config,
                    OptRef<xds::core::v3::ResourceLocator> odcds_resources_locator,
                    ProtobufMessage::ValidationVisitor& validation_visitor) PURE;
+
+  /**
+   * Calls getObject ObjectSharedPool to get sharedPtr of CommonLbConfig in pool
+   *
+   * @param common_lb_config The config field to be stored in ObjectSharedPool
+   * @return shared_ptr to the CommonLbConfig in ObjectSharedPool
+   */
+ virtual std::shared_ptr<const envoy::config::cluster::v3::Cluster::CommonLbConfig>
+ GetCommonLbConfigPtr(const envoy::config::cluster::v3::Cluster::CommonLbConfig common_lb_config) PURE;
```

Comment:
nit: Envoy cpp style guide is camel case for methods https://github.com/envoyproxy/envoy/blob/main/STYLE.md#:~:text=Function%20names%20should%20all%20use%20camel%20case%20starting%20with%20a%20lower%20case%20letter%20(e.g.%2C%20doFoo())